plugin-maven/src/main/java/com/diffplug/spotless/maven/ArtifactResolver.java

```diff
@@ -53,10 +56,18 @@ public ArtifactResolver(RepositorySystem repositorySystem, RepositorySystemSessi
 		this.log = Objects.requireNonNull(log);
 	}
 
+	/** Use {@link ArtifactResolver#resolve(Collection) instead.} */
+	@Deprecated
 	public Set<File> resolve(String mavenCoordinate) {
```

Comment:
@nedtwigg 
Remind me... For `lib` you are always strict about public interfaces since it can be used independent from `spotless` project. I assume that you use standard semantic versioning and remove some deprecated methods when you change the major version.

What about `maven-plugin`? Same approach?

Reply:
Because gradle scripts are programs, any public method in `lib`, `lib-extra`, or `plugin-gradle` might reasonably be called by somebody's build script, which is why I'm a stickler for those projects.  For `plugin-maven`, we can probably get away with being more loose, but when it's easy to keep back-compat, I think it's best to do so.  Just a good habit to be in for a project that aspires to help lots of people but cause few headaches.